tests/rados/test_osd_full.py

```diff
@@ -694,3 +689,158 @@ def check_full_warning(type, a_set):
                 "Verification of OSD host addition during rebalancing and cluster full has been completed"
             )
         return 0
+
+    if config.get("iops_with_full_osds"):
+        doc = (
+            "\n # CEPH-83572758 | BZ-2214864"
+            "\n Verify IOs are possbile on a cluster after one OSD reaches FULL state"
+            "\n\t 1. Create a replicated pool with single PG and autoscaling disabled"
+            "\n\t 2. Retrieve the acting set for pool 1"
+            "\n\t 3. Reweight the OSDs part of acting set of Pool 1 to 0"
+            "\n\t 4. Create another replicated pool and EC pool with single PG and autoscaling disabled"
+            "\n\t 5. Retrieve the acting set for pool 2 and 3, should not have any OSD from acting set of Pool 1"
+            "\n\t 6. Reweight the OSDs part of acting set of Pool 1 back to 1"
+            "\n\t 7. Perform IOPS using rados bench to fill up Pool 1 up till completely full capacity"
+            "\n\t 8. Perform IOPS on Pool 2 and Pool 3 and ensure IOPS is successful as full "
+            "OSDs are not part of acting set"
+        )
+
+        log.info(doc)
+        config = config["iops_with_full_osds"]
+        log.info("Running test for IOPs with cluster having full OSDs")
+
+        try:
+            rados_obj.configure_pg_autoscaler(**{"default": "off"})
+            pool1_name = pool_name = config["pool-1"]["pool_name"]
+            assert create_pool_per_config(**config["pool-1"])
+
+            # retrieving the size of each osd part of acting set for the pool
+            acting_set1 = rados_obj.get_pg_acting_set(pool_name=pool1_name)
+            match_osd_size(a_set=acting_set1)
+            primary_osd_size = osd_sizes[acting_set1[0]]
+
+            log.info(
+                f"All OSDs part of acting set {acting_set1} are of same size {primary_osd_size} KB"
+            )
+
+            # reweight osds to 0
+            [
+                cephadm.shell([f"ceph osd reweight osd.{osd_id} 0"])
+                for osd_id in acting_set1
+            ]
+
+            pool2_name = config["pool-2"]["pool_name"]
+            assert create_pool_per_config(**config["pool-2"])
+            acting_set2 = rados_obj.get_pg_acting_set(pool_name=pool2_name)
+
+            pool3_name = config["pool-3"]["pool_name"]
+            assert create_pool_per_config(**config["pool-3"])
+            acting_set3 = rados_obj.get_pg_acting_set(pool_name=pool3_name)
+
+            for osd_id in acting_set1:
```

Comment:
Please check once if the ` acting_set1 ` has changed over time, post marking OSD weight to 0.

Reply:
Added a check for this

Reply:
Post reweighting OSDs to 0, acting set will change, but will return to original state once OSDs are reweighted to 1
So the check has been added after reweight 1
Reference -
![image](https://github.com/red-hat-storage/cephci/assets/100675023/01692c6e-51ba-4bcc-b599-8cf280d8261b)

Reply:
The same result through automation -
![image](https://github.com/red-hat-storage/cephci/assets/100675023/f95d9c20-4636-4d15-b5f9-e37f469e72dc)